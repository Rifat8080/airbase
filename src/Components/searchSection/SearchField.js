import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink, useParams } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';
import { displayCountryMap } from '../../redux/Pollution/pollution.js';
import { fecthCountry } from '../../redux/Country/countries.js';
import search from './Search.module.css';
import map from '../../assets/map.png';

const SearchField = () => {
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.country_name);
  const { country } = countries;
  const { id } = useParams();
  const namePass = id ? id.toLowerCase() : 'europe';

  useEffect(() => {
    if (country.length === 0) {
      dispatch(fecthCountry(namePass));
    }
  }, []);

  const setDisplay = () => {
    dispatch(displayCountryMap(true));
  };

  const countryField = country.length !== 0
    ? country[0].map((field) => (
      <NavLink
        to={`/pollution/${field.latlag}:${field.name}:${field.code}`}
        className={search.nav}
        key={uuidv4()}
        onClick={setDisplay}
      >
        <div className={search.item_grid}>
          <div className={search.img_container}>
            <img src={map} alt="map" className={search.image_map} />
          </div>
          <div>
            {' '}
            <span className={search.field_name}>
              {' '}
              {field.name}
              {' '}
            </span>
            {' '}
            <div>
              {' '}
              Population:
              {field.population}
            </div>
            {' '}
          </div>
        </div>
      </NavLink>
    ))
    : ' ';

  return (
    <div className={search.contianer}>
      <h4 className={search.padh4} data-testid="search">
        {' '}
        STATES BY COUNTRY
        {' '}
      </h4>
      <div className={search.grid}>{countryField}</div>
    </div>
  );
};

export default SearchField;
