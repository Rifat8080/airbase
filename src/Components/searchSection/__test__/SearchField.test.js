import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import SearchField from '../SearchField.js';
import store from '../../../redux/configureStore.js';

test('SearchField snapshot test', () => {
  const component = renderer.create(
    <Provider store={store}>
      <SearchField />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
