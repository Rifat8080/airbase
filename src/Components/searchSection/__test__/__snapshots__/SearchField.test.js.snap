// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField snapshot test 1`] = `
<div
  className="contianer"
>
  <h4
    className="padh4"
    data-testid="search"
  >
     
    STATES BY COUNTRY
     
  </h4>
  <div
    className="grid"
  >
     
  </div>
</div>
`;
